### PR TITLE
Add FluentBit Cluster CRD

### PR DESCRIPTION
**Pull request by wenchajun**
Add FluentBit Cluster CRD such as clusterinput.fluentbit.fluent.io while keeping the original namespace crd like Input.logging.kubesphere.io
Signed-off-by: chengdehao <dehaocheng@yunify.com>

### CODE REVIEW

**Comment by wenchajun** [comment]
In this PR, I added ClusterFluentBitConfig,ClusterInput, ClusterOutput, ClusterParser and ClusterFilter CRDs. Users can select an appropriate Config for mounting. For example, If the field is set to `fluentBitConfigName: cluster-fluent-bit-config`, Select a cluster-level configuration on behalf of the user. If the field is set to `fluentBitConfigName: fluent-bit-config`, indicates that the namespace level is selected by the user.
 In addition to this, a cluster-level Secret is created, where a secret is created in either namespace. Please refer to
https://github.com/fluent/fluentbit-operator/pull/167/files#diff-0b752120e79a0e2d3a24281bbe4463ba6c9b115c9c0db57433160cb32fcf1f51R109

File: pkg/operator/daemonset.go [hunk 1]
```diff
@@ -96,7 +95,7 @@ func MakeScopedRBACObjects(fbName, fbNamespace string) (*rbacv1.Role, *corev1.Se
 	return &r, &sa, &rb
 }
 
-func MakeDaemonSet(fb v1alpha2.FluentBit, logPath string) appsv1.DaemonSet {
+func MakeDaemonSet(fb logging.FluentBit, logPath string) appsv1.DaemonSet {
```
Comment (zhu733756) [inline pkg/operator/daemonset.go:98]: I think v1alpha2 is more clear than logging.

File: apis/fluentbit.io/v1alpha2/clusterfilter_types.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	"bytes"
+	"fmt"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	plugins2 "kubesphere.io/fluentbit-operator/apis/plugins"
```
Comment (zhu733756) [inline apis/fluentbit.io/v1alpha2/clusterfilter_types.go:23]: can you use plugins instead?

File: apis/fluentbit.io/v1alpha2/clusterparser_types.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	"bytes"
+	"fmt"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	plugins2 "kubesphere.io/fluentbit-operator/apis/plugins"
```
Comment (zhu733756) [inline apis/fluentbit.io/v1alpha2/clusterparser_types.go:23]: There is no package with duplicate names here, so I think plugins can be used directly

File: cmd/manager/main.go [hunk 4]
```diff
@@ -118,6 +120,15 @@ func main() {
 		setupLog.Error(err, "unable to create controller", "controller", "FluentBit")
 		os.Exit(1)
 	}
+
+	if err = (&controllers.ClusterFluentBitConfigReconciler{
```
Comment (zhu733756) [inline cmd/manager/main.go:124]: Move up before the FluentBitReconciler is set up.

**Comment by benjaminhuo** [comment]
This is a big enhancement, I'll take a look at this PR @wenchajun

**Comment by wenchajun** [comment]
> This is a big enhancement, I'll take a look at this PR @wenchajun

thanks

File: cmd/manager/main.go (hunk 4 above)
Comment (wenchajun) [reply cmd/manager/main.go]: thanks，Most of this is auto-generated, and I've fixed it

**Review by wanjunlei — CHANGES_REQUESTED** [review]

**Comment by benjaminhuo** [comment]
@wanjunlei @zhu733756 Do you have any other comments on this PR?

**Comment by wenchajun** [comment]
Close this pr because [it](https://github.com/fluent/fluentbit-operator/pull/193) has been merged

**Pull request closed without merging**